spring-kafka/src/main/java/org/springframework/kafka/retrytopic/DefaultDestinationTopicResolver.java

```diff
@@ -197,6 +203,12 @@ public boolean isContextRefreshed() {
 		return this.contextRefreshed;
 	}
 
+	@Override
+	public void setApplicationContext(ApplicationContext applicationContext) {
+		this.applicationContext = applicationContext;
+		this.clock = applicationContext.getBean(RetryTopicInternalBeanNames.INTERNAL_BACKOFF_CLOCK_BEAN_NAME, Clock.class);
```

Comment:
You cannot call `getBean()` from the `setApplicationContext()` - too early, may cause some bean initialization problem...